File: jadwal_mobile/lib/models/wa_inbox_message.dart
class WaInboxMessage {
  WaInboxMessage({
    required this.id,
    required this.senderNumber,
    required this.senderName,
    required this.message,
    required this.receivedAt,
    required this.status,
    required this.replyMessage,
  });

  final int id;
  final String? senderNumber;
  final String? senderName;
  final String message;
  final String? receivedAt;
  final String status;
  final String? replyMessage;

  factory WaInboxMessage.fromJson(Map<String, dynamic> json) {
    return WaInboxMessage(
      id: (json['id'] as num?)?.toInt() ?? 0,
      senderNumber: json['sender_number'] as String?,
      senderName: json['sender_name'] as String?,
      message: (json['message'] as String?) ?? '',
      receivedAt: json['received_at'] as String?,
      status: (json['status'] as String?) ?? '',
      replyMessage: json['reply_message'] as String?,
    );
  }
}

File: app/Models/WaInboxMessage.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class WaInboxMessage extends Model
{
    public const STATUS_NEW = 'new';
    public const STATUS_ASSIGNED = 'assigned';
    public const STATUS_REPLIED = 'replied';

    protected $fillable = [
        'sender_number',
        'sender_name',
        'message',
        'received_at',
        'status',
        'assigned_to',
        'assigned_at',
        'replied_by',
        'replied_at',
        'reply_message',
        'meta',
    ];

    protected $casts = [
        'received_at' => 'datetime',
        'assigned_at' => 'datetime',
        'replied_at' => 'datetime',
        'meta' => 'array',
    ];

    public function assignedTo(): BelongsTo
    {
        return $this->belongsTo(Personil::class, 'assigned_to');
    }

    public function repliedBy(): BelongsTo
    {
        return $this->belongsTo(Personil::class, 'replied_by');
    }

    public static function statusOptions(): array
    {
        return [
            self::STATUS_NEW => 'Baru',
            self::STATUS_ASSIGNED => 'Diambil',
            self::STATUS_REPLIED => 'Dibalas',
        ];
    }
}
